Improve prefixing routes example in guides [ci-skip]

p8 opened the pull request:
### Summary

The current example is ambiguous as it uses 'admin' for both the scope
and `:as` option.
By using `helpdesk` for the scope the example becomes clearer.

skipkayhil commented:
Should the example just above the one you changed be changed as well? I think it could be similarly confusing

p8 commented:
@skipkayhil good point!
Now I'm hestitating if we should add a seperate example for the non-ambiguos example.
Or maybe just closing this PR.

skipkayhil commented:
> Now I'm hestitating if we should add a seperate example for the non-ambiguos example. Or maybe just closing this PR.

Hmm now that I'm reading the whole section I agree it may make sense to leave `scope 'admin'`... what if we update some of the wording to make it explicit what's being fixed? 

````diff
diff --git a/guides/source/routing.md b/guides/source/routing.md
index e1e92358fb..f4d32048d0 100644
--- a/guides/source/routing.md
+++ b/guides/source/routing.md
@@ -1143,7 +1143,10 @@ end
 resources :photos
 ```
 
-This will provide route helpers such as `admin_photos_path`, `new_admin_photo_path`, etc.
+This changes the route helpers for `/admin/photos` from `photos_path`,
+`new_photos_path`, etc. to `admin_photos_path`, `new_admin_photo_path`,
+etc. Without the addition of `as: 'admin_photos` on the scoped `resources
+:photos`, the non-scoped `resources :photos` will not have any route helpers.
 
 To prefix a group of route helpers, use `:as` with `scope`:
 
@@ -1155,7 +1158,9 @@ end
 resources :photos, :accounts
 ```
 
-This will generate routes such as `admin_photos_path` and `admin_accounts_path` which map to `/admin/photos` and `/admin/accounts` respectively.
+Just as before, this changes the `/admin` scoped resource helpers to
+`admin_photos_path` and `admin_accounts_path`, and allows the non-scoped
+resources to use `photos_path` and `accounts_path`.
 
 NOTE: The `namespace` scope will automatically add `:as` as well as `:module` and `:path` prefixes.
 ````

p8 commented:
@skipkayhil I've updated the PR and added you as a co-author.

skipkayhil commented on guides/source/routing.md:
````diff
@@ -1143,19 +1143,24 @@ end
 
 ```ruby
-scope 'admin', as: 'admin' do
+scope 'helpdesk', as: 'admin' do
````
Should this be reverted? (Or should we update the text to match this?)

p8 committed:
Improve prefixing routes example in guides [ci-skip]

The current example is ambiguous as it uses 'admin' for both the scope
and `:as` option.
This change clarifies the example by explaining what `scope` does and
what the `:as` option does.

Co-authored-by: Hartley McGuire <skipkayhil@gmail.com>

skipkayhil reviewed (approved):
Lgtm!

p8 commented:
Thanks @skipkayhil !
